src/builder.rs

```diff
@@ -0,0 +1,241 @@
+use super::Bytes;
+use super::BytesError;
+use super::Uuid;
+use super::Variant;
+use super::Version;
+
+/// A builder struct for creating a [`Uuid`]
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
```

Comment:
Let's remove the `Copy` implementation here, because the behaviour of `Copy` types and `&mut self` methods can be surprising.

Reply:
Oh interesting, I didn't know that this might cause weird behaviour. Will remove it then, this builder doesn't consume itself anyways so there is no need to copy it anyways.

Reply:
Actually removing the copy implementation will trigger the `missing_copy_implementations` lint.

Reply:
> Oh interesting, I didn't know that this might cause weird behaviour.

Yeh, the wierdness is that a type that implements `Copy` can be implicitly copied around, so when you're passing the value around and calling mutating methods you can end up mutating a copy by accident.

Reply:
Ah yes, that makes never though of it like that. That is good thing to keep in mind.